src/frontends/pytorch/src/transforms/dict_resolver.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "dict_resolver.hpp"
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/result.hpp"
+#include "openvino/op/util/framework_node.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace pass {
+
+using namespace ov::pass;
+using namespace ov::op;
+
+bool DictResolver::run_on_model(const std::shared_ptr<Model>& model) {
+    bool changed = false;
+    const auto results = model->get_results();
+    for (const auto& res : results) {
+        if (auto dict_construct_node = cast_fw_node(res->get_input_node_shared_ptr(0), "prim::DictConstruct")) {
+            const auto inputs = dict_construct_node->input_values();
+            if (inputs.size() % 2) {
+                // inputs must be divisible by 2
+                return false;
+            }
+            OutputVector new_outputs;
+            for (size_t i = 0; i < inputs.size(); i += 2) {
+                auto new_output = inputs.at(i + 1);
+                const auto name_node = inputs.at(i);
+                auto fw_node = std::dynamic_pointer_cast<ov::op::util::FrameworkNode>(name_node.get_node_shared_ptr());
+                if (!fw_node) {
+                    // odd inputs must contain constant strings encoded as fw nodes
+                    return false;
+                }
+                const auto attrs = fw_node->get_attrs();
+                if (attrs.find("string_value") == attrs.end()) {
+                    // fw node must contain string value
+                    return false;
+                }
+                auto name = attrs.at("string_value");
+                new_output.add_names({name});
+                new_outputs.push_back(new_output);
+            }
+            // TODO: this may break the order of outputs if after dict more outputs exist 
```

Comment:
What's the problem to provide fully functional implementation already now? I realize that when we 'unpacking' `dict` in such a way, this is already something that change the output signature, but looks not to too hard to maintain the remining part of outputs in the same order just to avoid changes without a good reason. As I remember we don't have a convenient way like ov::Model method, and in my experiments when I did such thing I added another method to ov::Model, but we can provide a helper function to make it without changing ov::Model API.

@itikhono, @ilyachur just FYI, a potential gap in ov::Model, or I'm just missing something.

Reply:
There is no `set_results` method for a `Model`, the only way to do it with current api is `remove_result` one by one and do `add_results` after.